### PR TITLE
Moe/decode

### PR DESCRIPTION
**Pull request by Moejoe90**

### CODE REVIEW

File: src/decoder.js [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
 
 /**
  * Decoding log that generated from the event
- * @param {*} log 
+ * @param {*} log
  */
-export const decodeLog = log => // eslint-disable-line 
-  // @todo: implement it
-  log;
+export const decodelogData = (logData, abi) => {
```
Comment (MostafaBalata) [inline src/decoder.js]: should be `decodeLogData`

File: src/decoder.js [hunk 2]
```diff
@@ -1,25 +1,29 @@
+import consensysDecode from 'abi-decoder';
+
 /**
  *
  * This will return get function name and parameters out
  * of the trasnaction input attribute if exists.
  *
  */
```
Comment (MostafaBalata) [inline src/decoder.js]: You should put the @params in the comments 💃

File: src/decoder.js [hunk 3]
```diff
@@ -1,25 +1,29 @@
...
-
-  return null;
+  throw new Error('Problem with input data check if data is sent correctly');
 };
 
 /**
  * Decoding log that generated from the event
- * @param {*} log 
+ * @param {*} log
  */
```
Comment (MostafaBalata) [inline src/decoder.js]: you forget this comment as well 💃

File: test/decoder.spec.js [hunk 4]
```diff
@@ -1,34 +1,49 @@
...
-            gasPrice: 123,
-            input: '0x57cb2fc4',
-        };
-    })
+  const esopjson = path.join(__dirname, 'mockedData', 'esopABI.json');
+  const abi = JSON.parse(fs.readFileSync(esopjson, { encoding: 'utf8' }));
+
+  it('should return function name and its params for the transaction', () => {
+    const dataSample1 = '0xcc96b9430000000000000000000000006a58a4e262b6ead9eb40f41313d6d1fbbbb41c160000000000000000000000000000000000000000000000000000000057fabde0000000000000000000000000000000000000000000000000000000005960cc0600000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001';
+    const openESOP = { name: 'offerOptionsToEmployee',
```
Comment (MostafaBalata) [inline test/decoder.spec.js]: the object formatting is not correct, can you use your editor to format the file again?

File: test/decoder.spec.js [hunk 5]
```diff
@@ -1,34 +1,49 @@
...
+    const dataSampleErr = 'a58a4e262b6ead9eb4';
+    expect(() => decodeInputData(dataSampleErr, abi)).to.throw('Problem with input data');
+  });
 
-    it('should return the function name and its params for the transaction', () => {    
-        delete transactionSample['input'];
-        const decodedInput = decodeInputData(transactionSample);
-        expect(decodedInput).to.be.equal(null);
-    });
+  it('should return function name and its params for the transaction', () => {
```
Comment (MostafaBalata) [inline test/decoder.spec.js]: I think you forget to change this text, because it's the same like the first one.

File: src/decoder.js (hunk 2 above)
Comment (Moejoe90) [reply src/decoder.js]: Done

File: src/decoder.js (hunk 3 above)
Comment (Moejoe90) [reply src/decoder.js]: Done

File: src/decoder.js (hunk 1 above)
Comment (Moejoe90) [reply src/decoder.js]: done

**Pull request merged**